### ISSUE TITLE
"nuclide-fuzzy-filename-provider" does not respect "Ignored Names" setting

### ISSUE THREAD

**Issue by billyvg** [facebook/nuclide#400]
### Issue and Steps to Reproduce
If "nuclide-fuzzy-filename-provider" feature is enabled, it causes Omnisearch to not respect the "Ignored Names" setting.

### Expected Behavior
It should not show files that are ignored

### Actual Behavior
It shows files that should be ignored

### Versions
* Atom: 1.5.3
* Nuclide: 0.120.0
* Client OS: OSX 10.11.3
* Server OS (optional):

### Additional Details
* Installed packages (`apm ls --installed`):

├── Stylus@3.0.0
├── ex-mode@0.8.0
├── haskell-grammar@0.4.0
├── highlight-selected@0.11.2
├── language-babel@2.15.5
├── language-ini@1.14.0
├── language-ocaml@1.1.2
├── language-thrift@1.0.2
├── linter@1.11.3
├── linter-eslint@7.0.0
├── merge-conflicts@1.3.7
├── nuclide@0.120.0
├── pigments@0.24.5
├── react@0.14.1
├── sort-lines@0.14.0
├── tool-bar@0.3.0
├── vim-mode@0.64.0
└── vim-surround@0.8.1

**Comment by thibmaek** [comment]
Yeah this is really annoying for dependency folders like node_modules/

**Comment by vanhtuan0409** [comment]
I hope this issue will be resolved soon

**Comment by courthead** [comment]
I fixed this by setting up my Atom config like so:

```cson
"*":
  core:
    ignoredNames: [
      "*/dist/**/*"
      "*/node_modules/**/*"
      "*/bower_components/**/*"
      "*/tmp/**/*"
    ]
```
